### ISSUE TITLE
[FEATURE] Interactively create an Index

### ISSUE THREAD

**Issue by Tejasmadhukar** [opensearch-project/opensearch-cli#94]
**Is your feature request related to a problem?**
Yes there has always been a little bit of hassle creating an Index either having to write the entire json or copy pasting it.
I was thinking of an easier way to make this where you can add mappings interactively.

**What solution would you like?**
I can to build a Terminal User Interface (TUI) through which users can interactively define mappings interactively and edit the json from there as they like.

**Comment by Tejasmadhukar** [comment]
@wbeckler

**Comment by wbeckler** [comment]
Could this be better handled in a GUI like OpenSearch Dashboards?

**Comment by Tejasmadhukar** [comment]
Yes you're right, It would work well in a user interface like one present in dashboards.
But I was thinking it would also be convenient to have something like this in the terminal.

https://github.com/opensearch-project/opensearch-cli/assets/39000441/7e1a53a8-a2b6-4843-b1f6-c4837099dab3

Users can define many mappings like this. (ignore the submit button)
And I think that it helpful to have this in terminal as a convenient way of defining index compared to dasboards where people have to first install then to use it.

**Comment by dblock** [comment]
Catch All Triage - [1](https://github.com/dblock) [2](https://github.com/dbwiddis) [3](https://github.com/krisfreedain) [4](https://github.com/akolarkunnu) [5](https://github.com/mingshl) [6](https://github.com/getsaurabh02)
